qiskit_experiments/calibration_management/basis_gate_library.py

```diff
@@ -0,0 +1,126 @@
+
+    # Location where default parameter values are stored. These may be updated at construction.
+    __default_values__ = dict()
+
+    def __init__(self, default_values: Optional[Dict] = None):
+        """Setup the library.
+
+        Args:
+            default_values: A dictionary to override library default parameter values.
+        """
+        self._schedules = dict()
+
+        if default_values is not None:
+            for key, value in default_values.items():
+                self.__default_values__[key] = value
+
+    def __getitem__(self, name: str) -> ScheduleBlock:
+        """Return the schedule."""
+        return self._schedules[name]
+
+    @abstractmethod
+    def default_values(self) -> List[Tuple[ParameterValueType, Parameter, Tuple, ScheduleBlock]]:
+        """Return the default values for the parameters.
+
+        Returns
+            A list of tuples is returned. These tuples are structured so that instances of
+            :class:`Calibrations` can call :meth:`add_parameter_value` on the tuples.
+        """
+
+
+class FixedFrequencyTransmonSXRZ(BasisGateLibrary):
+    """A library of gates for fixed-frequency superconducting qubit architectures."""
+
+    __default_values__ = {"duration": 160, "amp": 0.5, "β": 0.0}
+
+    def __init__(self, default_values: Optional[Dict] = None):
+        """Setup the schedules."""
+        super().__init__(default_values)
+
+        dur = Parameter("duration")
+        sigma = Parameter("σ")
+
+        # X gates
+        sched_x = self._single_qubit_schedule("x", dur, Parameter("amp"), sigma, Parameter("β"))
+
+        # square-root X gates
+        sched_sx = self._single_qubit_schedule("sx", dur, Parameter("amp"), sigma, Parameter("β"))
+
+        for sched in [sched_x, sched_sx]:
+            self._schedules[sched.name] = sched
+
+    @staticmethod
+    def _single_qubit_schedule(
+        name: str,
+        dur: Parameter,
+        amp: Parameter,
+        sigma: Parameter,
+        beta: Optional[Parameter] = None,
+    ) -> ScheduleBlock:
+        """Build a single qubit pulse."""
+
+        chan = pulse.DriveChannel(Parameter("ch0"))
+
+        if beta is not None:
+            with pulse.build(name=name) as sched:
+                pulse.play(pulse.Drag(duration=dur, amp=amp, sigma=sigma, beta=beta), chan)
+        else:
+            with pulse.build(name=name) as sched:
+                pulse.play(pulse.Gaussian(duration=dur, amp=amp, sigma=sigma), chan)
+
+        return sched
+
+    def default_values(self) -> List[Tuple[ParameterValueType, Parameter, Tuple, ScheduleBlock]]:
+        """Return the default values for the parameters.
+
+        Returns
+            A list of tuples is returned. These tuples are structured so that instances of
+            :class:`Calibrations` can call :meth:`add_parameter_value` on the tuples.
+        """
+        defaults = []
+        for name in ["x", "sx"]:
```

Comment:
I think RZ should be provided otherwise we need to separately add RZ if we start from scratch. Also this allows us to easily migrate to frames.

Reply:
I'm leaning towards leaving the RZ out of this as there is no schedule to calibrate.

Reply:
I see. We cannot distinguish the parameter to calibrate from the parameters of gates?

Reply:
Since there is nothing to calibrate in the RZ gate it should not be in the calibrations.